commit 12da14a59135fe331f294b91488c1bdf55b35b4e
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Wed Nov 26 12:54:39 2014 +0000

    Fix warnings in fwscanf / rewind tests.
    
    This patch fixes "set but not used" warnings in the tests
    libio/bug-rewind.c and libio/bug-rewind2.c by checking the return
    values from fwscanf and making the tests fail if those values aren't
    as expected.
    
    Tested for x86_64.
    
            * libio/bug-rewind.c (do_test): Check fwscanf return values.
            * libio/bug-rewind2.c (do_test): Likewise.

diff --git a/libio/bug-rewind.c b/libio/bug-rewind.c
index 4f8242df3d6d1f75..1734de285f30fff4 100644
--- a/libio/bug-rewind.c
+++ b/libio/bug-rewind.c
@@ -52,9 +52,19 @@ do_test (void)
 
   rewind (fptr);
   ret1 = fwscanf (fptr, L"%c%c", &arg1, &arg2);
+  if (ret1 != 2)
+    {
+      printf ("first fwscanf returned %d, expected 2\n", ret1);
+      return 3;
+    }
 
   rewind (fptr);
   ret2 = fwscanf (fptr, L"%c%n%c", &arg1, &num, &arg2);
+  if (ret2 != 2)
+    {
+      printf ("second fwscanf returned %d, expected 2\n", ret2);
+      return 4;
+    }
 
   if (arg2 != 'd')
     {
diff --git a/libio/bug-rewind2.c b/libio/bug-rewind2.c
index 51b574433029f835..496f628675b36fcc 100644
--- a/libio/bug-rewind2.c
+++ b/libio/bug-rewind2.c
@@ -45,6 +45,11 @@ do_test (void)
 
       rewind (fp);
       ret = fwscanf (fp, L"%c", dummy);
+      if (ret != 1)
+	{
+	  printf ("fwscanf returned %d, expected 1\n", ret);
+	  result = 1;
+	}
 
       printf ("current pos = %ld\n", ftell (fp));
       if (ftell (fp) != 1)
